sql/chunk_constraint.sql

```diff
@@ -0,0 +1,174 @@
+CREATE OR REPLACE FUNCTION _timescaledb_internal.create_chunk_constraint(
+    chunk_id INTEGER,
+    constraint_oid OID
+)
+    RETURNS VOID LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    sql_code    TEXT;
+    constraint_row pg_constraint;
+    chunk_row _timescaledb_catalog.chunk;
+BEGIN
+    SELECT * INTO STRICT constraint_row FROM pg_constraint WHERE OID = constraint_oid;
+    SELECT * INTO STRICT chunk_row FROM _timescaledb_catalog.chunk WHERE id = chunk_id;
+
+    INSERT INTO _timescaledb_catalog.chunk_constraint (hypertable_constraint_name, chunk_id, constraint_name)
+    SELECT constraint_row.conname, chunk_row.id, format('%s_%s_%s', chunk_id,  nextval('_timescaledb_catalog.chunk_constraint_name'), constraint_row.conname);
+END
+$BODY$;
+
+-- Drop a constraint on a chunk
+-- static
+CREATE OR REPLACE FUNCTION _timescaledb_internal.drop_chunk_constraint(
+    chunk_id INTEGER,
+    constraint_name NAME
+)
+    RETURNS VOID LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    chunk_row _timescaledb_catalog.chunk;
+BEGIN
+    DELETE FROM _timescaledb_catalog.chunk_constraint c
+    WHERE c.hypertable_constraint_name = drop_chunk_constraint.constraint_name 
+    AND c.chunk_id = drop_chunk_constraint.chunk_id;
+END
+$BODY$
+SET client_min_messages = WARNING;
+
+-- do I need to add a hypertable constraint to the chunks?;
+CREATE OR REPLACE FUNCTION _timescaledb_internal.need_chunk_constraint(
+    hypertable_id INTEGER,
+    constraint_oid OID
+)
+    RETURNS BOOLEAN LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+    constraint_row record;
+BEGIN
+    SELECT * INTO STRICT constraint_row FROM pg_constraint WHERE OID = constraint_oid;
+
+    IF constraint_row.contype IN ('c') THEN
+        -- check and not null constraints handled by regular inheritance (from docs):
+        --    All check constraints and not-null constraints on a parent table are automatically inherited by its children,
+        --    unless explicitly specified otherwise with NO INHERIT clauses. Other types of constraints
+        --    (unique, primary key, and foreign key constraints) are not inherited."
+
+        IF constraint_row.connoinherit THEN
+            RAISE 'NO INHERIT option not supported on hypertables: %', constraint_row.conname
+            USING ERRCODE = 'IO101';
+        END IF;
+
+        RETURN FALSE;
+    END IF;
+    RETURN TRUE;
+END
+$BODY$;
+
+-- Creates a constraint on all chunk for a hypertable.
+CREATE OR REPLACE FUNCTION _timescaledb_internal.add_constraint(
+    hypertable_id INTEGER,
+    constraint_oid OID
+)
+    RETURNS VOID LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+    constraint_row pg_constraint;
+    hypertable_row _timescaledb_catalog.hypertable;
+BEGIN
+    IF _timescaledb_internal.need_chunk_constraint(hypertable_id, constraint_oid) THEN
+        SELECT * INTO STRICT constraint_row FROM pg_constraint WHERE OID = constraint_oid;
+        IF constraint_row.conindid <> 0 AND constraint_row.contype != 'f' THEN
```

Comment:
Does 'f' mean foreign? I'd like to see this commented or made into a function.

Reply:
Fixed (In PR #198)